server/src/sockets/socket.utils.js

```diff
@@ -0,0 +1,58 @@
+
+class Game {
+  constructor(socketId) {
+    this.player1 = {
+      socketId,
+      color: 'w',
+    };
+    this.inPlay = false;
+  }
+
+  addPlayer2(socketId) {
+    this.player2 = {
+      socketId,
+      color: 'b',
+    };
+    this.inPlay = true;
+    return this;
+  }
+}
+
+class Games {
+  constructor() {
+    this.live = [];
+  }
+
+  addGame(socketId) {
+    const game = new Game(socketId);
+    this.live.push(game);
+    return game;
+  }
+
+  idleGame() {
```

Comment:
does this return Boolean? Name is not very intuitive to what it does.

Reply:
This function returns the first Game object that is in idle state in the list of all games. If no idle game is found it returns undefined.

Reply:
please name it as findIdleGame or something in that case.

Reply:
Done